src/ExportTemplates/EVCPlugin.luau

```diff
@@ -540,20 +540,28 @@ for _, func in pairs(moduleStore:GetChildren()) do
 						}
 						tweenThreads[lightName] = faderTable
 						tweenThreads[lightName].thread = task.defer(function()
-							while task.wait(1 / 3) do
-								if faderTable.run == false then
+							while task.wait(1/3) do
+								if faderTable.run == false or lights[lightName].running_module ~= patternModule then
 									break
-								elseif lights[lightName].running_module == patternModule and lightParts[lightName] then
+								else
 									if faderTable.currentRunner >= rawlen(lightData) then
 										faderTable.currentRunner = 1
 									else
 										faderTable.currentRunner += 1
 									end
 
-									local lightData: faderLight = lightData[faderTable.currentRunner]
+									local lightData:faderLight = lightData[faderTable.currentRunner]
 									if lightData.color ~= 0 then
-										lightParts[lightName].Color =
-											patternModule.moduleSettings.colors[lightData.color]
+										if lightParts[lightName]:FindFirstChild("Lighto") then
```

Comment:
Consider doing a `local lighto = lightParts[lightName]:FindFirstChildWhichIsA("SurfaceGui")` and then reusing that variable in all places like this.